platform/mime/core/platform-mime-core-impl/src/main/java/ddf/mime/mapper/MimeTypeMapperImpl.java

```diff
@@ -175,6 +181,25 @@ public String guessMimeType(InputStream is, String fileExtension)
         // from being used when a CustomMimeTypeResolver may be more appropriate.
         List<MimeTypeResolver> sortedResolvers = sortResolvers(mimeTypeResolvers);
 
+        if (StringUtils.isEmpty(fileExtension)) {
+            try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+                IOUtils.copy(is, baos);
```

Comment:
You can't do this. This will blow up the heap if someone is attempting to upload a very large file. You need to use a TemporaryFileBackedOutputStream which can fail over to the file system in the event that the InputStream has too many bytes in it.